### ISSUE TITLE
Automation ID locator strategy using accessibility ID is not working for winappdriver

### ISSUE THREAD

**Issue by praveendvd** [webdriverio/webdriverio#7174]
**Environment (please complete the following information):**
 - **WebdriverIO version:** [e.g. 4.13.2] : 7.8.0
 - **Mode:** [Standalone mode or WDIO Testrunner] : Standalone
 - **If WDIO Testrunner, running sync/async:** [e.g. sync/async] : async
 - **Node.js version:** [e.g. 8.11.2] : v16
 - **NPM version:** [e.g. 5.8.0] v6
 - **Browser name and version:** [e.g. Chrome 68] 
 - **Platform name and version:** [e.g. Windows 10] :win10
 - **Additional wdio packages used (if applicable):** [e.g. @wdio/spec reporter, @wdio/selenium-standalone service]

**Config of WebdriverIO**

**Describe the bug**

As per appium documentation:

![image](https://user-images.githubusercontent.com/45095911/126860234-df8275d7-2e56-44a0-a5fb-bb24ea282bf2.png)

But in webdriverIO,

non of the below locators except xpath works

```
$(".someCLass")
$("~automationID") etc
```

this throws invalid location strategy

but i can use : 

```
await $(await browser.findElement("accessibility id","someautomationid" ))
```

**To Reproduce**

[wdio.conf.zip](https://github.com/webdriverio/webdriverio/files/6872034/wdio.conf.zip)

**Expected behavior**:

Appium locators should work with webdriverio locators

**Comment by christian-bromann** [comment]
Can you provide some logs?

**Comment by praveendvd** [comment]
[logs.txt](https://github.com/webdriverio/webdriverio/files/6877112/logs.txt)

@christian-bromann  is this enough?

**Comment by praveendvd** [comment]
```

const findStrategy = function (selector, isW3C, isMobile) {
    const stringSelector = selector;
    let using = DEFAULT_STRATEGY;
    let value = selector;
    switch (defineStrategy(selector)) {
        // user has specified locator strategy directly
        case 'directly': {
            const match = stringSelector.match(DIRECT_SELECTOR_REGEXP);
            if (!match || !isMobile && isW3C && !constants_1.W3C_SELECTOR_STRATEGIES.includes(match[1])) {
                throw new Error('InvalidSelectorStrategy'); // ToDo: move error to wdio-error package
            }
            using = match[1];
            value = match[2];
            break;
        }
        case 'xpath': {
            using = 'xpath';
            break;
        }
        case 'id': {
            using = 'id';
            value = stringSelector.slice(3);
            break;
        }
        case 'link text': {
            using = 'link text';
            value = stringSelector.slice(1);
            break;
        }
        case 'partial link text': {
            using = 'partial link text';
            value = stringSelector.slice(2);
            break;
        }
        case 'shadow':
            using = 'shadow';
            value = stringSelector.slice(constants_1.DEEP_SELECTOR.length);
            break;
        case '-android uiautomator': {
            using = '-android uiautomator';
            value = stringSelector.slice(8);
            break;
        }
        case '-android datamatcher': {
            using = '-android datamatcher';
            value = JSON.stringify(value);
            break;
        }
        case '-android viewmatcher': {
            using = '-android viewmatcher';
            value = JSON.stringify(value);
            break;
        }
        case '-ios uiautomation': {
            using = '-ios uiautomation';
            value = stringSelector.slice(4);
            break;
        }
        case 'accessibility id': {
            using = 'accessibility id';
            value = stringSelector.slice(1);
            break;
        }
        case 'class name': {
            using = 'class name';
            break;
        }
        case 'tag name': {
            using = 'tag name';
            value = stringSelector.replace(/<|>|\/|\s/g, '');
            break;
        }
        case 'name': {
            if (isMobile || !isW3C) {
                const match = stringSelector.match(/^\[name=("|')([a-zA-z0-9\-_.@=[\] ']+)("|')]$/);
                if (!match) {
                    throw new Error(`InvalidSelectorMatch. Strategy 'name' has failed to match '${stringSelector}'`);
                }
                using = 'name';
                value = match[2];
            }
            break;
        }
        case 'xpath extended': {
            using = 'xpath';
            const match = stringSelector.match(new RegExp(XPATH_SELECTOR_REGEXP.map(rx => rx.source).join('')));
            if (!match) {
                throw new Error(`InvalidSelectorMatch: Strategy 'xpath extended' has failed to match '${stringSelector}'`);
            }
            const PREFIX_NAME = { '.': 'class', '#': 'id' };
            const conditions = [];
            const [tag, prefix, name, attrName, attrValue, partial, query] = match.slice(1);
            if (prefix) {
                conditions.push(`contains(@${PREFIX_NAME[prefix]}, "${name}")`);
            }
            if (attrName) {
                conditions.push(attrValue
                    ? `contains(@${attrName}, "${attrValue}")`
                    : `@${attrName}`);
            }
            conditions.push(partial ? `contains(., "${query}")` : `normalize-space() = "${query}"`);
            value = `.//${tag || '*'}[${conditions.join(' and ')}]`;
            break;
        }
        case '-image': {
            using = '-image';
            value = fs_1.default.readFileSync(stringSelector, { encoding: 'base64' });
            break;
        }
        case 'role': {
            const match = stringSelector.match(/^\[role=(.+)\]/);
            if (!match) {
                throw new Error(`InvalidSelectorMatch. Strategy 'role' has failed to match '${stringSelector}'`);
            }
            using = 'css selector';
            value = createRoleBaseXpathSelector(match[1]);
            break;
        }
    }
    /**
     * ensure selector strategy is supported
     */
    if (!isMobile && isW3C && (using !== 'shadow') && !constants_1.W3C_SELECTOR_STRATEGIES.includes(using)) {
        console.log("hsdasdasdasdsssssssssssssssssssssssssssss",constants_1,using)
        throw new Error('InvalidSelectorStrategy'); // ToDo: move error to wdio-error package
    }
    return { using, value };
};

````


this is taking isMobile as fals and checking the locator to be in the w3c strategy

**Comment by christian-bromann** [comment]
It would suggest to remove the if statement and have the browser driver return an error if a selector strategy is not supported.

**Comment by christian-bromann** [comment]
Fixed by #7183
